src/test/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperationsTest.scala

```diff
@@ -473,6 +474,120 @@ class TezosDatabaseOperationsTest
       exists shouldBe false
 
     }
+
+    "get map from a block table" in {
+      implicit val randomSeed = RandomSeed(testReferenceTime.getTime)
+
+      val blocks = generateBlockRows(1, testReferenceTime)
+      val columns = List("level", "proto", "protocol", "hash")
+
+      val populateAndTest = for {
+        _ <- Tables.Blocks ++= blocks
+        found <- sut.selectWithPredicates(Tables.Blocks.baseTableRow.tableName, columns, List.empty)
+      } yield found
+
+      val result = dbHandler.run(populateAndTest.transactionally).futureValue
+      result shouldBe List(
+        Map("level" -> 0, "proto" -> 1, "protocol" -> "protocol", "hash" -> "R0NpYZuUeF"),
+        Map("level" -> 1, "proto" -> 1, "protocol" -> "protocol", "hash" -> "aQeGrbXCmG")
+      )
+    }
+
+    "get map from a block table with predicate" in {
+      implicit val randomSeed = RandomSeed(testReferenceTime.getTime)
+
+      val blocks = generateBlockRows(1, testReferenceTime)
+      val columns = List("level", "proto", "protocol", "hash")
+      val predicates = List(
+        Predicates(
+          field = "hash",
+          operation = OperationType.in,
+          set = List("R0NpYZuUeF"),
+          inverse = false
+        )
+      )
+
+      val populateAndTest = for {
+        _ <- Tables.Blocks ++= blocks
+        found <- sut.selectWithPredicates(Tables.Blocks.baseTableRow.tableName, columns, predicates)
+      } yield found
+
+      val result = dbHandler.run(populateAndTest.transactionally).futureValue
+      result shouldBe List(
+        Map("level" -> 0, "proto" -> 1, "protocol" -> "protocol", "hash" -> "R0NpYZuUeF")
+      )
+    }
+
+    "get map from a block table with inverse predicate" in {
+      implicit val randomSeed = RandomSeed(testReferenceTime.getTime)
+
+      val blocks = generateBlockRows(1, testReferenceTime)
+      val columns = List("level", "proto", "protocol", "hash")
+      val predicates = List(
+        Predicates(
+          field = "hash",
+          operation = OperationType.in,
+          set = List("R0NpYZuUeF"),
```

Comment:
As before

Reply:
as described above - pseudo random generator with hardcoded seed gives every time the same values.

Reply:
hardcoded